src/variable_info_detail.hpp

```diff
@@ -16,80 +16,150 @@
 
 #pragma once
 
-#include <string>
 #include "config.hpp"
+#include "utils/const_clone.hpp"
+
+#include <string>
+#include <type_traits>
 
-namespace variable_info_detail
+class invalid_variablename_exception : public std::exception
```

Comment:
If you need to throw invalid_variable_name exception, you can always organize variable_info.hpp like so:

```
#include stuff

class invalid_variable_name_exception public std::exception { ... };

#include "variable_info_detail.hpp"

... the rest of the header ...
```